### PR TITLE
Remove usage of clouds.yaml file to openstack connection

### PR DESCRIPTION
**Pull request by yhaliaw**
Applicable spec: n/a

### Overview

Remove the usage of clouds.yaml file to create openstack connection.

Also fixes a bug introduced in https://github.com/canonical/github-runner-manager/pull/21 (MetricsStorage that is not hashable has been used in a set comprehension).

Integration + E2E tests: https://github.com/canonical/github-runner-operator/actions/runs/11374351617

### Rationale

This prevents the multi-processing fighting over the file handle for the clouds.yaml file.  

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] The library version in `pyproject.toml` is incremented

### CODE REVIEW

File: src/github_runner_manager/openstack_cloud/openstack_runner_manager.py [hunk 1]
```diff
@@ -410,11 +401,11 @@ def _cleanup_extract_metrics(
         # On the other hand, there could be storage for runners from the past that
         # should be cleaned up.
         all_runner_names = healthy_runner_names | unhealthy_runner_names
-        unmatched_metrics_storage = {
+        unmatched_metrics_storage = (
             ms
             for ms in metrics_storage_manager.list_all()
             if ms.runner_name not in all_runner_names
-        }
+        )
```
Comment (cbartz) [inline src/github_runner_manager/openstack_cloud/openstack_runner_manager.py:408]: This fixes a bug because MetricsStorage is not hashable.

**Review by cbartz — APPROVED** [review]
I took over the PR and made some minor changes, so I think I am "allowed" to approve it.

**Review by jdkandersson — APPROVED** [review]

File: src/github_runner_manager/openstack_cloud/openstack_cloud.py [hunk 2]
```diff
@@ -29,19 +28,36 @@
...
+        region_name: The region.
+        user_domain_name: The domain name containing the user.
+        project_domain_name: The domain name containing the project.
+    """
+
+    auth_url: str
+    project_name: str
+    username: str
+    password: str
+    region_name: str
```
Comment (yanksyoon) [inline src/github_runner_manager/openstack_cloud/openstack_cloud.py:56]: The region name belongs to the Auth level
```
clouds:
  testcloud:
    auth:
      auth_url: ...
      password: ,,,
      project_domain_name: ...
      project_name: ...
      user_domain_name: ...
      username: ...
    region_name: <REGION_NAME_HERE>
```

would this be suitable here?

Comment (yanksyoon) [reply src/github_runner_manager/openstack_cloud/openstack_cloud.py:56]: I think it should be okay since we don't use clouds.yaml file anymore

**Review by yanksyoon — APPROVED** [review]
LGTM!

File: src/github_runner_manager/openstack_cloud/openstack_cloud.py (hunk 2 above)
Comment (cbartz) [reply src/github_runner_manager/openstack_cloud/openstack_cloud.py:56]: I think its ok, we are not modelling the clouds.yaml here.

**Pull request merged**